db/compaction/compaction_iterator.cc

```diff
@@ -251,6 +251,8 @@ bool CompactionIterator::InvokeFilterIfNeeded(bool* need_skip,
           valid_ = false;
           return false;
         }
+        // TODO(ajkr): What happens when this runs in flush/recovery, i.e., when
```

Comment:
@ltamasi would be good to get your input here. I can investigate a bit more but am leaning toward setting a non-ok status and returning false in this case, unless anyone has a different suggestion.

Reply:
This logic pertains to the integrated BlobDB implementation and can be only hit during compaction, since large values are not yet written to blob files at this point during flush/recovery. (That happens in `ExtractLargeValueIfNeeded`, which is called from `PrepareOutput`.) So I think what you suggested would work :)

Reply:
Thanks!

Reply:
> setting a non-ok status and returning false in this case

Done.